src/H5LS.c

```diff
@@ -75,10 +80,19 @@ const H5LS_mmap_class_t *get_H5LS_mmap_class_t(char *type) {
     p = &H5LS_GPU_mmap_ext_g;
 #endif
   } else {
-    LOG_ERROR(-1,
-              "I don't know the type of storage: %s\n"
-              "Supported options: SSD|BURST_BUFFER|MEMORY|GPU\n",
-              type);
+
+    size_t copy_len = strlcpy(truncated_msg, type, sizeof(truncated_msg));
```

Comment:
Do we care too much about using non-standard C?

Reply:
Yes, I fixed to use snprintf